src/dag/transitive.js

```diff
@@ -0,0 +1,104 @@
+const {NotFoundGraphError} = require('graphology');
+
+/**
+ * Graphology Transitive DAG operation
+ * ====================================
+ *
+ * Function that takes a cartesian product of a node's in-edges with
+ * its out-edges and performs an operation on each pair.
+ *
+ * This function is m*n in time and space, where m is the node's in-degree
+ * and n is the node's out-degree.
+ */
+function forEachTransitiveRelation(graph, node, callback) {
```

Comment:
I am not sure this function should consider this fact but what happens if you encounter self loops?

Reply:
I think this implementation would work correctly on self loops and even on cyclic or undirected graphs, but I haven't thought deeply about that. It just iterates over pairs of edges. As written, one of the iterated edge pairs would just be the same edge twice.